File: bdp-master-server/src/main/java/com/github/bdp/master/server/service/ServerService.java
package com.github.bdp.master.server.service;

import java.util.List;
import com.github.bdp.master.server.domain.Server;

public interface ServerService {

	Server findOne(Long id);

	Server findOne(String serverName);

	void save(Server server);

	void delete(Long id);

	List<Server> findAll();

	void loadAll();
}

File: bdp-master-client/src/main/scala/com/github/bdp/master/client/service/ServerService.scala
package com.github.bdp.master.client.service

import com.github.bdp.master.client.Constants._
import com.github.bdp.master.client.domain.Server
import com.github.bdp.master.client.util.{JsonDecoder, RedisClient}
import com.typesafe.scalalogging.LazyLogging

object ServerService extends LazyLogging {

  def getServerBy(id: Long): Server = {
    JsonDecoder.decodeServer(RedisClient.get(s"$SERVER_KEYSPACE:$id"))
  }

  def getServerBy(hostname: String): Server = {
    val key = RedisClient.get(s"i_$SERVER_KEYSPACE:$hostname")
    JsonDecoder.decodeServer(RedisClient.get(key))
  }

}